lib/Serialization/ModuleFile.cpp

```diff
@@ -297,22 +298,36 @@ namespace {
 /// Used to deserialize entries in the on-disk decl hash table.
 class ModuleFile::DeclTableInfo {
 public:
-  using internal_key_type = StringRef;
-  using external_key_type = Identifier;
+  using internal_key_type = std::pair<DeclNameKind, StringRef>;
```

Comment:
This is good, use something like this for the ClangImporter lookup tables, please.